protocol/packet_test.go

```diff
@@ -1,25 +1,28 @@
-package protocol
+package protocol_test
```

Comment:
Why did you change the package name

Reply:
I think its a good convention to start doing it since that way we will only test methods/functions which are exposed which should help with creating feature tests instead of implementation tests. Besides that packages with _test behind will be build in their own binary. so I didnt really saw a downside to do it.

Reply:
Nice, I wasn't aware of the binary part